org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -9823,32 +9823,43 @@ private char[] capitalize(char[] name) {
 		return result;
 	}
 
-	private char[] findParameterNameAtLocationFromAssistParent(ObjectVector methodsFound) {
+	private char[] findParameterNameAtLocationFromAssistParent(ObjectVector methodsFound, Scope scope) {
 		char[] parameterName = new char[0];
 		MethodBinding[] candidates;
-		final MessageSend parent;
-		if (this.parser.assistNodeParent instanceof MessageSend) {
-			parent = (MessageSend) this.parser.assistNodeParent;
-			candidates = Optional.ofNullable(parent.actualReceiverType)
-					.or(() -> Optional.ofNullable(parent.receiver).map(r -> r.resolvedType))
-					.map(t -> t.getMethods(parent.selector)).orElse(new MethodBinding[0]);
-		} else if (this.parser.assistNode instanceof MessageSend) {
-			parent = (MessageSend) this.parser.assistNode;
+		final Expression[] arguments;
+		if (this.parser.assistNodeParent instanceof MessageSend ms) {
+			arguments = ms.arguments;
+			candidates = Optional.ofNullable(ms.actualReceiverType)
+					.or(() -> Optional.ofNullable(ms.receiver).map(r -> r.resolvedType))
+					.map(t -> t.getMethods(ms.selector)).orElse(new MethodBinding[0]);
+		} else if (this.parser.assistNode instanceof MessageSend ms) {
+			arguments = ms.arguments;
 			candidates = StreamSupport.stream(methodsFound.spliterator(), false)
 					.filter(Objects::nonNull)
 					.filter(o -> o instanceof Object[]).map(o -> (Object[]) o)
 					.map(o -> o[0]).filter(o -> o instanceof MethodBinding).map(b -> (MethodBinding) b)
-					.filter(b -> CharOperation.equals(parent.selector, b.selector)).findFirst()
+					.filter(b -> CharOperation.equals(ms.selector, b.selector)).findFirst()
 					.map(m -> new MethodBinding[] { m }).orElse(new MethodBinding[0]);
+		} else if (this.parser.assistNodeParent instanceof AllocationExpression ae) {
+			arguments = ae.arguments;
+			if (ae.type != null) {
+				ReferenceBinding binding = (ReferenceBinding) ae.type.resolvedType;
```

Comment:
do we need to use `instanceof` to check if the resolved type is a ReferenceBinding?

Reply:
Good point, it could be null as well. So I will update the code.